plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -179,8 +179,9 @@ data object LegacyUnspecifiedFunctionTypeEmbedding : UnspecifiedFunctionTypeEmbe
 
 data class FunctionTypeEmbedding(val signature: CallableSignatureData) : UnspecifiedFunctionTypeEmbedding() {
     override val name = object : MangledName {
+        // TODO: this can cause some number of collisions; fix it if it becomes an issue.
         override val mangled: String =
-            "fun_take\$${signature.formalArgTypes.joinToString("$") { it.name.mangled }}\$return\$${signature.returnType.name.mangled}"
+            "TF\$${signature.formalArgTypes.joinToString("$") { it.name.mangled }}"
```

Comment:
Why do we want this `TF` if we already have `F` before in functions? It feels like after introduced abbreviations this `T` is less readable than `take` before.

Reply:
Because this is a type, so this makes it consistent with our other types like `T_Int`.  That said, I'm not entirely happy with this either...